server.js

```diff
@@ -11,13 +13,78 @@ const welcomeMessage = {
   text: "Welcome to CYF chat system!",
 };
 
-//This array is our "data store".
-//We will start with one message in the array.
-//Note: messages will be lost when Glitch restarts our server.
 const messages = [welcomeMessage];
 
 app.get("/", function (request, response) {
   response.sendFile(__dirname + "/index.html");
 });
 
-app.listen(process.env.PORT);
+//Read all messages
+app.get("/messages", function (request, response) {
+  response.json(messages);
+});
+
+//Read only the most recent 10 messages: /messages/latest
+app.get("/messages/latest", function (request, response) {
+  let result = messages.filter(
+    (message, index) => index > messages.length - 11
+  );
+  response.json(result);
+});
+
+//Read only messages whose text contains a given substring: /messages/search?text=express
+app.get("/messages/search", function (request, response) {
+  const searchText = request.query.text.toLocaleLowerCase();
+  let result = messages.filter((message) =>
+    message.text.toLocaleLowerCase().includes(searchText)
+  );
+  response.json(result);
+});
+
+// Read one message specified by an ID
+app.get("/messages/:messageID", function (request, response) {
+  const messageID = request.params.messageID;
+  if (messageID) {
+    const message = messages.filter((message) => message.id == messageID);
+    if (message.length != 0) {
```

Comment:
As you are given a unique ID and are only expecting 1 (or zero) results, find() is a better method to use here. You can also return a 404 status code if the message is not found.

Reply:
Thank you for your comment. It is very useful